pgxn/neon/libpagestore.c

```diff
@@ -282,38 +388,32 @@ pageserver_disconnect(void)
 	 * time later after we have already sent a new unrelated request. Close
 	 * the connection to avoid getting confused.
 	 */
-	if (connected)
+	if (page_servers[shard_no].conn)
 	{
-		neon_log(LOG, "dropping connection to page server due to error");
-		PQfinish(pageserver_conn);
-		pageserver_conn = NULL;
-		connected = false;
+		neon_shard_log(shard_no, LOG, "dropping connection to page server due to error");
+		PQfinish(page_servers[shard_no].conn);
+		page_servers[shard_no].conn = NULL;
 
 		prefetch_on_ps_disconnect();
```

Comment:
Hmm, so if the connection to any pageserver is lost, we throw away the whole prefetch queue, even for other pageservers? I guess that's OK from a performance point of view: connection loss is supposed to be a rare event. Some comments to point that out and justify it would be good.

Reply:
Added comment.